### DISCUSSION TITLE
option about whether to copy on short tap or not

### DISCUSSION THREAD

**Discussion by corobin** [ente-io/ente#1029] [Enhancements]
android app

**Is your feature request related to a problem? Please describe.**
currently, an entry is copied when either:

- a single short tap
- a single long tap

is made on the main entry

**Describe the solution you'd like**
an option to configure whether a short tap triggers a copy or not

**Describe alternatives you've considered**
n/a

**Additional context**
2 rationale:

1. just that some users may have a preference one way or another

3. if #331 is implemented, then a short tap may be the trigger for revealing a code, and in that event it may be more important that a user wants to reveal the code but NOT copy it to the un-protected device clipboard. before that is implemented, a user who does not want to copy to clipboard can just not tap on an entry at all

**Comment by slubman** [comment]
I was about to make a similar feature request.

The single tap to reveal is used in a lot of other authenticators, and coming to Ente, I most of the time instinctively single tap and get frustrated.

Could it be implemented as an option to configure for each of the following what I prefer:
* Code reveal: single press, double press, long press (perhaps even triple press?)
* Code copy: single press, double press, long press (perhaps even triple press?)

This way, the user can adapt the application to his way of thinking.

**Reply by corobin** [reply]
> Could it be implemented as an option to configure

of course, the common trap with "just make it an option" to avoid having to resolve conflicting user preferences is that developers now have exponentially more configurations to support

however, I do think that in this case a single tap to reveal should be the preferred method, with an option for user choice if the developers want to support that

because you're right: single tap to reveal is the standard among authenticator apps, and ente auth is the odd one out, so it's going against conventional habit.
